src/index.js

```diff
@@ -300,39 +335,32 @@ module.exports = (
             );
 
             if (cleanBody !== undefined) {
-              Object.keys(cleanBody).forEach(name => {
-                // We neither have an easy way to transform `name` into `name[]` to signify that it's an array payload
-                // (and for all we know it might be an array of objects!), but also at the same time the HAR spec
-                // doesn't give any guidance for these kinds of cases so instead we're just falling back to
-                // stringifying the content instead of potentially including `fileName` properties.
-                if (Array.isArray(cleanBody[name])) {
-                  har.postData.params.push({
-                    name,
-                    value: JSON.stringify(cleanBody[name]),
-                  });
+              const multipartParams = multipartBodyToFormatterParams(
+                formData.body,
+                operation.schema.requestBody.content['multipart/form-data']
+              );
 
-                  return;
-                }
+              Object.keys(cleanBody).forEach(name => {
+                const param = multipartParams.find(multipartParam => multipartParam.name === name);
 
-                const data = {
-                  name,
-                  value: String(cleanBody[name]),
-                };
+                const value = formatter(formData, param, 'body', true);
 
                 // If we're dealing with a binary type, and the value is a valid data URL we should parse out any
                 // available filename and content type to send along with the parameter to interpreters like `fetch-har`
                 // can make sense of it and send a usable payload.
+                const addtlData = {};
+
                 if (binaryTypes.includes(name)) {
-                  const parsed = parseDataUrl(data.value);
+                  const parsed = parseDataUrl(value);
```

Comment:
I'm **really** worried about running a potential data URL through the `formatter` and then trying to parse it out.

Reply:
todo: add tests for data urls

Reply:
I realized no one is going to apply a style to a binary format, so this shouldn't be an issue. If they do, we'll do our best, and probably treat it like a string.